Lib/Src/CUartCom.cpp

```diff
@@ -109,7 +109,10 @@ bool CUartCom::send(etl::string<MAX_STRING_SIZE> msg)
     // Add message to queue
     if (m_tx_queue.size() <= MAX_TX_QUEUE_SIZE && (msg.empty() == false))
     {
-        m_tx_queue.put(msg);
+        if (m_tx_queue.put(msg) == false)
+        {
+            send("Error: Buffer overflow -> TX Queue\n");
```

Comment:
In this `send()` method you are finding that Tx queue is full and then call `send()` method to put more stuff in Tx queue.

Reply:
Why not make buffers so big you cannot saturate them sending reasonable amount of data?

Reply:
Suggestion, have a coms status flag (enums)
COMS_OK,
COMS_BUSY,
COMS_OVERFLOW etc 
When overflow happens set the status to COMS_OVERFLOW either wait for it clear or disallow any new messages to be sent. 

In the past, I used SDO to push error messages and kept the serial UART clean.